app/services/friends_service.rb

```diff
@@ -0,0 +1,17 @@
+class FriendsService
+  class << self
+    def invite(emails)
+      @incorrect_emails = ''
+      emails = emails.select do |email|
+        if email[/\A([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})\z/i]
+          User.exists?(email: email) ? false : true
+        else
+          @incorrect_emails += @incorrect_emails.blank? ? email : (', ' + email)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.